library/ssl_tls13_client.c

```diff
@@ -949,9 +1121,8 @@ static int ssl_tls13_parse_server_hello( mbedtls_ssl_context *ssl,
      */
     if( ssl_tls13_check_server_hello_session_id_echo( ssl, &p, end ) != 0 )
     {
-        MBEDTLS_SSL_PEND_FATAL_ALERT( MBEDTLS_SSL_ALERT_MSG_ILLEGAL_PARAMETER,
-                                      MBEDTLS_ERR_SSL_ILLEGAL_PARAMETER );
-        return( MBEDTLS_ERR_SSL_ILLEGAL_PARAMETER );
+        fatal_alert = MBEDTLS_ERR_SSL_ILLEGAL_PARAMETER;
```

Comment:
As the local variable got `alert` in its name please rather use the TLS 1.3 alert mnemonics here and in the other places.
Thus here, rather:
```C
fatal_alert = MBEDTLS_SSL_ALERT_MSG_ILLEGAL_PARAMETER;
```
Otherwise, we could just use `ret` and Mbed TLS error codes here but `ret` is used to store the return value of the functions called by this function. This should be ok but we would have more checks to do and this would be on the side of a bigger restructuring of the error handling. I'd say better not to start that here.

Reply:
So, will we change the fatal alert to the alert type to align with TLS1.3 mnemonics and keep the variable fatal_alert temp?

Reply:
I was suggesting for this PR to have here:
```C
fatal_alert = MBEDTLS_SSL_ALERT_MSG_ILLEGAL_PARAMETER;
```
and the same in similar places.